http.c

```diff
@@ -3719,6 +3768,13 @@ evhttp_set_allowed_methods(struct evhttp* http, ev_uint16_t methods)
 	http->allowed_methods = methods;
 }
 
+void
+evhttp_set_extended_methods(struct evhttp* http,
+    const struct evhttp_extended_methods *ext_methods)
+{
+	http->ext_methods = ext_methods;
```

Comment:
We need to copy, instead of accepting `const` here (this will reduce number of bugs I hope, and besides we already do this as a more-or-less generic rule, see `evhttp_connection_set_timeout_tv()` for example)